README.md

```diff
@@ -94,134 +147,145 @@ if let j: AnyObject = json {
+`NSJSONSerialization`.
+
+The `JSONValue` enum exists to help with some of the type inference, and also
+wraps up some of the casting that we'll need to to to transform the JSON into
+native types.
 
-Then the decoding implementation will look like this:
+Next, you need to make sure that models that you wish to decode from JSON
+conform to the `JSONDecodable` protocol:
 
 ```swift
-extension User: JSONDecodable {
-  static func create(id: Int)(name: String)(email: String?) -> User {
-    return User(id: id, name: name, email: email)
-  }
+public protocol JSONDecodable {
+  typealias DecodedType = Self
+  class func decode(JSONValue) -> DecodedType?
+}
+```
 
-  static func decode(j: JSONValue) -> User? {
-    return User.create
-      <^> j <|  "id"
-      <*> j <|  "name"
-      <*> j <|? "email"
+You will need to implement the `decode` function to perform any kinds of
+transformations you need to transform your model from a JSON value. A simple
+implementation for an enum value might look like:
+
+```swift
+enum RoleType: String {
+  case Admin = "Admin"
+  case User = "User"
+}
+
+extension RoleType: JSONDecodable {
+  static func decode(j: JSONValue) -> RoleType? {
+    switch j {
+    case let .JSONString(s): return RoleType(rawValue: s)
+    default: return .None
     }
   }
 }
 ```
 
-We create the user by calling `User.create` and use fmap (`<^>`) and apply
-(`<*>`) to check if each value exists within the `JSONValue` object. If any
-value is missing, the operation will return `.None`; otherwise, we'll receive
-the `User`. `j` is our `JSONValue` object and we pull values from it by using
-the `<|` operator (`<|?` for optionals) along with the key that references the
-value we want. It's important that these values follow the same order as the
-`create` function parameters.
+The real power of Argo can be seen when decoding actual model objects. To
+illustrate this, we will decode the simple `User` object that we used earlier.
 
-## Advanced
-
-As long as your models conform to `JSONDecodable` you can use them in other
-models like this `Post` model.
+Create your model normally:
 
 ```swift
-struct Post {
+struct User {
   let id: Int
-  let text: String
-  let author: User
+  let name: String
 }
+```
 
-extension Post: JSONDecodable {
-  static func create(id: Int)(text: String)(author: User) -> Post {
-    return Post(id: id, text: text, author: author)
-  }
+You will also want to create a curried creation function. This is needed
+because of a deficiency in Swift that doesn't allow us to pass normal `init`
+functions around as normal functions.
 
-  static func decode(j: JSONValue) -> Post? {
-    return Post.create
-      <^> j <| "id"
-      <*> j <| "text"
-      <*> j <| "author"
-    }
+```swift
+extension User {
+  static func create(id: Int)(name: String) -> User {
+    return User(id: id, name: name)
   }
 }
 ```
 
-From the JSON:
+Using this curried syntax will allow us to partially apply the function over
```

Comment:
I think that's explained in the linked article, but I can make it more explicit that it's explained there.